File: python/calculate_mode.py
def calculate_mode(array_of_elements):
    """Calculate the mode from a given array of elements

    Args:
        array_of_elements (array): An array of elements

    Returns:
        array: An array of elements that are the mode from the given array
    """

    # This is a sorted list of each unique element containing it's key/value
    list_of_elements = sorted([[x, array_of_elements.count(x)] for x in set(array_of_elements)])

    # List of modes to return
    list_of_modes = []

    # Grab the highest value
        # (sorted numbers don't sort correctly so it'll grab the lowest 'key' value 
        # (where key is the given element from the array))
    mode_val = list_of_elements[0][1]

    # Loop through the list, reversed to catch all cases
    for nested_data in reversed(list_of_elements):
        # is the current data value higher or equal to the mode val?
        if nested_data[1] >= mode_val:
            # Add to the list of modes, and set the mode_val to our val since we are the highest
            list_of_modes.append(nested_data[0])
            mode_val = nested_data[1]
    # Return the list of modes
    return print(list_of_modes)

calculate_mode(["a", "b", "c", "b", "b", "a", "a"])
calculate_mode([1,3,3])

File: js/calculateMode.js
/**
 * Calcuate the mode with any given array and return all modes in an array
 * @param {array} arrayOfElements Given array to evaluate the mode for
 * @returns an array of all modes within the given array
 */
const calculateMode = (arrayOfElements) => {
    /**
     * @param answerKey This is our object that we use to store all unique entries of our given array
     */
    const answerKey = {};

    // Loop through each element within the given array
    for (let idx in arrayOfElements) {
        // At each element we add it to our object and increment the value
        answerKey[arrayOfElements[idx]] = (answerKey[arrayOfElements[idx]] || 0) + 1;
    }

    /**
     * @param nestedAnswerKey Take the object of unique elements from the given array, and make it an array of key/values within an array
     */
    const nestedAnswerKey = Object.entries(answerKey);

    /**
     * @param sortedAnswerKey An inline sort function since this particular sorting doesn't require special logic for sort
     */
    const sortedAnswerKey = nestedAnswerKey.sort((a, b) => a - b);

    /**
     * @param mode Array of modes from the given array
     */
    let mode = [];

    /**
     * @param modeVal We grab the value from the first mode to determine if there are more mode elements within the array
     */
    const modeVal = sortedAnswerKey[0][1];

    // Loop through the array
    for (nestedElement in sortedAnswerKey) {
        // As soon as the element's value is not equal to the mode's value we stop iterating
        if (sortedAnswerKey[nestedElement][1] != modeVal) {
            break;
        }

        // This element's value equals the modeVal so add it to the mode array
        mode.push(sortedAnswerKey[nestedElement][0]);
    }

    // Return the mode array which contains all mode elements from the given array
    return console.log(mode);
}

// Test cases
calculateMode([5, 6, 7, 8]);
calculateMode(["6", "6", "7", "8", "9", "10", "18", "28"]);